gamemode/core/libs/cl_markup.lua

```diff
@@ -22,444 +22,490 @@ TEXT_ALIGN_TOP		= 3
-		elseif ( tag == "font" or tag == "face" ) then
-			table.remove( font_stack )
+local function ExtractParams(p1,p2,p3)
+
+	if (string.utf8sub(p1, 1, 1) == "/") then
+		
+		local tag = string.utf8sub(p1, 2)
+		
+		if (tag == "color" or tag == "colour") then
+			table.remove(colour_stack)			
+		elseif (tag == "font" or tag == "face") then
+			table.remove(font_stack)
 		end
 
 	else
 
-		if ( p1 == "color" or p1 == "colour" ) then
-
-			local rgba = colourMatch( p2 )
-
-			if ( rgba == nil ) then
+		if (p1 == "color" or p1 == "colour") then
+			
+			local rgba = colourMatch(p2)
+			
+			if (rgba == nil) then
 				rgba = {}
 				local x = { "r", "g", "b", "a" }
-				local n = 1
-				for k, v in string.gmatch( p2, "(%d+),?" ) do
-					rgba[ x[ n ] ] = k
+				n = 1
+				for k, v in string.gmatch(p2, "(%d+),?") do
+					rgba[ x[n] ] = k
 					n = n + 1
 				end
 			end
+						
+			table.insert(colour_stack, rgba)
+		
+		elseif (p1 == "font" or p1 == "face") then
 
-			table.insert( colour_stack, rgba )
+			table.insert(font_stack, tostring(p2))
+		elseif (p1 == "img" and p2) then
+			local exploded = string.Explode(",", p2)
+			local material = exploded[1] or p2
+			local p3 = exploded[2]
 
-		elseif ( p1 == "font" or p1 == "face" ) then
+			local found = file.Find("materials/"..material..".*", "GAME")
 
-			table.insert( font_stack, tostring( p2 ) )
+			if (found[1] and found[1]:find("%.png")) then
+				material = material..".png"
+			end
 
+			local texture = Material(material)
+			local sizeData = string.Explode("x", p3 or "16x16")
+			w = tonumber(sizeData[1]) or 16
+			h = tonumber(sizeData[2]) or 16
+
+			if (texture) then
+				table.insert(blocks, {
+					texture = texture,
+					w = w,
+					h = h
+				})
+			end
 		end
-
+		
 	end
 end
 
 --[[---------------------------------------------------------
-	Name: CheckTextOrTag(p)
-	Desc: This function places data in the "blocks" table
-		  depending of if p is a tag, or some text
-	Usage: ** INTERNAL ** Do not use!
+    Name: CheckTextOrTag(p)
+    Desc: This function places data in the "blocks" table
+          depending of if p is a tag, or some text
+   Usage: ** INTERNAL ** Do not use!
 -----------------------------------------------------------]]
-local function CheckTextOrTag( p )
-	if ( p == "" ) then return end
-	if ( p == nil ) then return end
-
-	if ( string.utf8sub( p, 1, 1 ) == "<" ) then
-		string.gsub( p, "<([/%a]*)=?([^>]*)", ExtractParams )
+local function CheckTextOrTag(p)
+	if (p == "") then return end
+	if (p == nil) then return end
+	
+	if (string.utf8sub(p, 1, 1) == "<") then
+		string.gsub(p, "<([/%a]*)=?([^>]*)", ExtractParams)
 	else
-
+		
 		local text_block = {}
-		text_block.text = p
-		text_block.colour = colour_stack[ #colour_stack ]
-		text_block.font = font_stack[ #font_stack ]
-		table.insert( blocks, text_block )
-
+		text_block.text = p 
+		text_block.colour = colour_stack[ table.Count(colour_stack) ]
+		text_block.font = font_stack[ table.Count(font_stack) ]
```

Comment:
```suggestion
		text_block.font = font_stack[ #font_stack ]
```
As pointed out by Heyter, # is even better here